notify/impl.go

```diff
@@ -62,6 +66,85 @@ func Build(confs []*config.Receiver, tmpl *template.Template) map[string]Fanout
 
 const contentTypeJSON = "application/json"
 
+// TemplateData is the data passed to notification templates.
+// End-users should not be exposed to Go's type system,
+// as this will confuse them and prevent simple things like
+// simple equality checks to fail. Map everything to float64/string.
+type TemplateData struct {
+	Status            string
+	Alerts            []TemplateAlert
+	AlertCommonLabels map[string]string
+
```

Comment:
I don't think the use case is as strong as for labels, I can see potential use but only in a similar way to what I'm doing to subject which won't tend to have space to fit a summary. I figure a wait and see approach is best, we can always add it in later.